projects/packages/forms/src/contact-form/class-contact-form-endpoint.php

```diff
@@ -853,18 +858,32 @@ private function get_plugin_status( $plugin_slug ) {
 			'details'         => array(),
 		);
 
-		// Plugin-specific customizations
-		if ( 'akismet' === $plugin_slug ) {
-			$dashboard_view_switch                         = new Dashboard_View_Switch();
-			$response['isConnected']                       = class_exists( 'Jetpack' ) && \Jetpack::is_akismet_active();
-			$response['details']['formSubmissionsSpamUrl'] = $dashboard_view_switch->get_forms_admin_url( 'spam' );
-			$response['needsConnection']                   = true;
-		} elseif ( 'zero-bs-crm' === $plugin_slug && $is_active ) {
-			$has_extension       = function_exists( 'zeroBSCRM_isExtensionInstalled' ) && zeroBSCRM_isExtensionInstalled( 'jetpackforms' ); // @phan-suppress-current-line PhanUndeclaredFunction -- We're checking the function exists first
-			$response['details'] = array(
-				'hasExtension'         => $has_extension,
-				'canActivateExtension' => current_user_can( 'manage_options' ),
-			);
+		// Refactored to use a switch statement for plugin-specific logic.
+		switch ( $plugin_slug ) {
+			case 'akismet':
+				$dashboard_view_switch                         = new Dashboard_View_Switch();
+				$response['isConnected']                       = class_exists( 'Jetpack' ) && \Jetpack::is_akismet_active();
+				$response['details']['formSubmissionsSpamUrl'] = $dashboard_view_switch->get_forms_admin_url( 'spam' );
+				$response['needsConnection']                   = true;
+				break;
+			case 'zero-bs-crm':
+				if ( $is_active ) {
+					$has_extension       = function_exists( 'zeroBSCRM_isExtensionInstalled' ) && zeroBSCRM_isExtensionInstalled( 'jetpackforms' ); // @phan-suppress-current-line PhanUndeclaredFunction -- We're checking the function exists first
+					$response['details'] = array(
+						'hasExtension'         => $has_extension,
+						'canActivateExtension' => current_user_can( 'manage_options' ),
+					);
+				}
+				break;
+			case 'mailpoet':
+				$response['needsConnection'] = true;
+				if ( class_exists( '\MailPoet\Config\ServicesChecker' ) ) {
+					$checker = new \MailPoet\Config\ServicesChecker(); // @phan-suppress-current-line PhanUndeclaredClassMethod -- we're checking the class exists first
+					if ( method_exists( $checker, 'isMailPoetAPIKeyValid' ) ) {
+						$response['isConnected'] = (bool) $checker->isMailPoetAPIKeyValid( false ); // @phan-suppress-current-line PhanUndeclaredClassMethod -- we're checking the method exists first
+					}
+				}
+				break;
```

Comment:
I've added phan exceptions above. Phan will throw errors when calling classes/methods from outside a repo/codebase, even if wrapped in class_exists or method_exists conditionals. 

Known phan issue: https://github.com/phan/phan/issues/1204

We could also work around this by adding a stub, but that seems excessive. If anyone has concerns about adding these exceptions, let me know.